std/algorithm/searching.d

```diff
@@ -2330,6 +2330,31 @@ if (isForwardRange!R1 && isForwardRange!R2
     assert(findSkip(s, "def") && s.empty);
 }
 
+struct FindSplitResult(R1, R2)
+    if (isForwardRange!R1 &&
+        isForwardRange!R2)
+{
+    this(R1 pre, R1 separator, R2 post)
+    {
+        asTuple = typeof(asTuple)(pre, separator, post);
+    }
+
+    Tuple!(R1, "pre", R1, "separator", R2, "post") asTuple;
+
+    bool opCast(T : bool)()
+    {
+        return !asTuple.separator.empty;
+    }
+
+    alias asTuple this;
+}
+
+/// Instantiator for FindSplitResult
+auto findSplitResult(R1, R2)(R1 pre, R1 separator, R2 post)
```

Comment:
I don't think this should be public. There should never be a need for the user to create instances of `FindSplitResult`; they can already use `Tuple` in these cases.